File: doc-examples/example-python/src/test/java/example/support/FakeKeyVaultKeySigner.java
package example.support;

import com.azure.security.keyvault.keys.cryptography.models.SignatureAlgorithm;
import io.micronaut.azure.secretmanager.signing.KeyVaultKeySigner;
import jakarta.inject.Singleton;
import org.jspecify.annotations.NonNull;

import java.nio.charset.StandardCharsets;

/**
 * Stands in for the Key Vault backed signer, which needs a vault, and records the key and algorithm in the signature.
 */
@Singleton
public class FakeKeyVaultKeySigner implements KeyVaultKeySigner {

    public static final SignatureAlgorithm DEFAULT_ALGORITHM = SignatureAlgorithm.ES256;

    @Override
    public byte[] sign(@NonNull String keyName, @NonNull SignatureAlgorithm algorithm, @NonNull byte[] data) {
        return (keyName + ":" + algorithm + ":" + new String(data, StandardCharsets.UTF_8)).getBytes(StandardCharsets.UTF_8);
    }

    @Override
    public byte[] sign(@NonNull String keyName, @NonNull byte[] data) {
        return sign(keyName, DEFAULT_ALGORITHM, data);
    }
}

File: doc-examples/example-kotlin/src/test/kotlin/example/support/FakeKeyVaultKeySigner.kt
package example.support

import com.azure.security.keyvault.keys.cryptography.models.SignatureAlgorithm
import io.micronaut.azure.secretmanager.signing.KeyVaultKeySigner
import jakarta.inject.Singleton

/**
 * Stands in for the Key Vault backed signer, which needs a vault, and records the key and algorithm in the signature.
 */
@Singleton
class FakeKeyVaultKeySigner : KeyVaultKeySigner {

    override fun sign(keyName: String, algorithm: SignatureAlgorithm, data: ByteArray): ByteArray =
        "$keyName:$algorithm:${String(data)}".toByteArray()

    override fun sign(keyName: String, data: ByteArray): ByteArray = sign(keyName, SignatureAlgorithm.ES256, data)
}
